tests/unit-tests/core/main-class.php

```diff
@@ -75,4 +75,15 @@ public function test_wc_class_instances() {
 		$this->assertInstanceOf( 'WC_Payment_Gateways', $this->wc->payment_gateways() );
 		$this->assertInstanceOf( 'WC_Checkout', $this->wc->checkout() );
 	}
+	
+	public function test_user_agent_header( $webhook = '' ) {
+		if ( ! empty( $webhook ) ) {
+        	$this->expectOutputString( 'WooCommerce/3.5.3 Hookshot (compatible; ClassicCommerce/1.0.0)' );
+        	print 'WooCommerce/' . WC()->version . ' ' . $webhook . '(compatible; ClassicCommerce/' . WC()->cc_version . ')';
+		}
+		else {
+			$this->expectOutputString( 'WooCommerce/3.5.3 (compatible; ClassicCommerce/1.0.0)' );
+        	print 'WooCommerce/' . WC()->version . ' (compatible; ClassicCommerce/' . WC()->cc_version . ')';
```

Comment:
It's better to use one of the "regular" assertion functions like `$this->assertEquals` rather than a `print` function. `$this->expectOutputString` is really meant for when you need to test a more complicated piece of code that happens to print output, and you want to test that the output is as expected but you can't easily change the way that other code works.